examples/stupid/stupid08-write-data-to-file.py

```diff
@@ -4,79 +4,101 @@
-                         new_t.y*model.height/model.y_tiles)
-            self.align()
-        self.draw_color()
+        # Does nothing, if all tiles are occupied
+
+    def eat(self, model):
+        # Eat from the current tile
+        tile = self.current_tile()
+        self.grow_size += min(model["max_food_eat"], tile.info["food"])
+        tile.info["food"] = max(0, tile.info["food"]-model["max_food_eat"])
+        self.size_to_color()
+
+    def step(self, model):
+        self.eat(model)
+        self.move()
+        if self.grow_size > 100:
+            model.pause()
 
 
 def setup(model):
+    # Open data file for writing
     global f
     f = open("stupid.data", "w")
+
     model.reset()
-    people = set([Bug() for i in range(math.floor(model["initial_bugs"]))])
-    model["stop"] = False
-    model.add_agents(people)
-    for t in model.tiles:
-        t.info["food"] = 0.0
-        t.color = (0, 0, 0)
+
+    # Add agents
+    for i in range(int(model["initial_bugs"])):
+        model.add_agent(Bug())
+
+    # Initialize tiles
+    for tile in model.tiles:
+        tile.info["food"] = 0.0
+        tile.color = (0, 0, 0)
 
 
 def step(model):
     global f
-    if not model["stop"]:
-        bug_min = None
-        bug_mean = 0
-        bug_max = None
-        for a in model.agents:
-            a.step(model)
-            if not bug_min or bug_min > a.grow_size:
-                bug_min = a.grow_size
-            if not bug_max or bug_max < a.grow_size:
-                bug_max = a.grow_size
-            bug_mean += a.grow_size
-        bug_mean /= model["initial_bugs"]
-        f.write(str(bug_min) + " " + str(bug_mean) + " " + str(bug_max) + "\n")
-        f.flush() # Flush is necessary as long as we can't call f.close()
-                  # when the user exits the program
-
-        for t in model.tiles:
-            food_prod = random.random() * model["max_food_prod"]
-            t.info["food"] += food_prod
-            c = min(255, math.floor(t.info["food"] * 255))
-            t.color = (c, c, c)
-        model.update_plots()
+
+    # Food production
+    for tile in model.tiles:
+        food_prod = random.uniform(0, model["max_food_prod"])
+        tile.info["food"] += food_prod
+        c = min(255, math.floor(tile.info["food"] * 255))
+        tile.color = (c, c, c)
+
+    # Move all agents
+    for agent in model.agents:
+        agent.step(model)
+
+    # Calculate min, average and max bug size
+    bug_min = 100
+    bug_sum = 0
+    bug_max = 0
+    for agent in model.agents:
+        bug_min = min(bug_min, agent.grow_size)
+        bug_max = max(bug_max, agent.grow_size)
+        bug_sum += agent.grow_size
+    bug_mean = bug_sum / len(model.agents)
+
+    # Write min, average and max bug size to file
+    f.write(str(bug_min) + " " + str(bug_mean) + " " + str(bug_max) + "\n")
+    # Flush is necessary as long as we can't call f.close()
+    # when the user exits the program
+    f.flush()
```

Comment:
This should not be necessary anymore, as we have the `Model.on_close` method where a call to `f.close` can be added.

Reply:
oh yeah, will look at that next time I'm working on this PR (you are also welcome to add commits to this branch though)